enumerable.rb

```diff
@@ -0,0 +1,350 @@
+    else
+      i = 0
+      arr = []
+      while i < self.size
+        yield self[i]
+        if yield self[i]
+          arr << self[i]
+        end
+        i += 1
+      end
+      arr
+    end
+  end
+
+  def my_all?(word = nil)
+    unless block_given?
+      i = 0
+      j = 0
+
+      while i < self.size
+        self[i]
+        i += 1
+      end
+      if self.size == 0
+        true
+      elsif word.class == Regexp
+
+        for element in self
+          if element.match(word)
+            j += 1
+          end
+        end
+        j == self.length ? true : false
+
+      elsif word == Numeric
+
+        for element in self
+          if element.class <= word
+            j += 1
+          end
+        end
+        j == self.length ? true : false
+      elsif word != nil
+        i = 0
+        while i < self.size
+          if self[i - 1] == self[i]
+            j += 1
+          end
+          i += 1
+        end
+
+        for element in self
+          if element.class == word
+            j += 1
+          end
+        end
+        j == self.length ? true : false
+
+      else
+        for element in self do
+          if element == false or element == nil
+            k += 1
+          end
+        end
+        k == 0 ? true : false
+      end
+    else
+      i = 0
+      j = 0
+      while i < self.size
+        yield self[i]
+        i += 1
+      end
+
+      for element in self
+
+        if yield(element) == false or yield(element) == nil
+          j += 1
+        end
+      end
+      j == 0 ? true : false
+
+    end
+  end
+
+  def my_any?(word = nil)
+    unless block_given?
+      i = 0
+      j = 0
+      while i < self.size
+        self[i]
+        i += 1
+      end
+      if self.size == 0
+        false
+      elsif word.class == Regexp
+
+        for element in self
+          if element.match(word)
+            j += 1
+          end
+        end
+        j > 0 and j <= self.length ? true : false
+
+      elsif word != nil
+        i = 0
+        k = 0
+        while i < self.size
+          if self[i - 1] == self[i]
+            k += 1
+          end
+          i += 1
+        end
+
+        for element in self
+          if element.class == word
+            k += 1
+          end
+        end
+        k > 0 ? true : false
+
+      elsif word == Numeric
+        for element in self
+          if element.class <= word
+            j += 1
+          end
+        end
+
+        j > 0 and j <= self.length ? true : false
+
+      else
+        k = 0
+        for element in self do
+          if element == false or element == nil
+            k += 1
+          end
+        end
+        k != self.size ? true : false
+
+      end
+
+    else
+      i = 0
+      j = 0
+      while i < self.size
+        yield self[i]
+        i += 1
+      end
+
+      for element in self
+        if yield element
+          j += 1
+        end
+      end
+      j > 0 and j <= self.size ? true : false
+    end
+  end
```

Comment:
#### my_any

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

    > %w[dog door rod blade].my_any?('cat') should be equal to %w[dog door rod blade].any?('cat') # false